shared/testgrid/yaml.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testgrid
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+
+	"github.com/knative/test-infra/shared/common"
+	"gopkg.in/yaml.v2"
+)
+
+const configPath = "ci/testgrid/config.yaml"
+
+// AllConfig is entire testgrid config
+type AllConfig struct {
+	Dashboards []Dashboard `yaml:"dashboards"`
+}
+
+// Dashboard is single dashboard on testgrid
+type Dashboard struct {
+	Name string `yaml:"name"`
+	Tabs []Tab  `yaml:"dashboard_tab"`
+}
+
+// Tab is a single tab on testgrid
+type Tab struct {
+	Name          string `yaml:"name"`
+	TestGroupName string `yaml:"test_group_name"`
+}
+
+// NewConfig loads from default config
+func NewConfig() (*AllConfig, error) {
+	root, err := common.GetRootDir()
+	if nil != err {
+		return nil, err
+	}
+	return NewConfigFromFile(path.Join(root, configPath))
+}
+
+// NewConfigFromFile loads config from file
+func NewConfigFromFile(fp string) (*AllConfig, error) {
+	ac := &AllConfig{}
+	contents, err := ioutil.ReadFile(fp)
+	if nil == err {
+		err = yaml.Unmarshal(contents, ac)
+	}
+	return ac, err
+}
+
+// GetTabRelURL finds URL relative to testgrid home URL from testgroup name
+// (generally this is prow job name)
+func (ac *AllConfig) GetTabRelURL(tgName string) (string, error) {
```

Comment:
```suggestion
func (ac *AllConfig) GetTabURL(tgName string) (string, error) {
```

Reply:
I would prefer the old name since it is really a relative URL, we may need to add GetTabURL function in the future if we need the full URL